Documentation/developers/debugging.rst

```diff
@@ -193,7 +193,186 @@ If you use CLion, you can debug Polycube with the following steps:
 3. At this point there are two equivalent ways to debug Polycube with CLion:
 
    a. Use the CLion debugger: at the top right select Debug 'polycube';
-   b. (or) Run Polycube in a terminal like that ``sudo polycubed --loglevel = DEBUG`` in order to enable the Polycube Debug logs and then from CLion, go to Run->Attach to Process.. and search for "polycubed"
+   b. (or) Run Polycube in a terminal with enabled debug logs (e.g., ``sudo polycubed --loglevel = DEBUG``); then, from CLion, go to Run->Attach to Process.. and search for "polycubed"
+
+4. At this point, from another terminal, just use ``polycubectl`` to interact with Polycube.
+
+
+
+Debugging network traffic using tcpdump/Wireshark
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Debugging (i.e., sniffing) network traffic frlowing *between* two Polycube services (e.g., on the link that connects a Polycube router to a Polycube bridge) can be achieved with the :ref:`span mode <span-mode>`.
+Please refer to that section for more details and for some working examples.
+
+
+Let's see how to capture packets using the concepts of shadow port and :ref:`span mode <span-mode>` to see what happens. 
+The topology used is a simpler version of the shadow type test3 of the router service, in order to understand well the various components, how to use tcpdump/Wireshark and understand what happens.
+
+
+::             	
+
+			pcn-r1 (network namespace)
+		      +-----------------------------+
+   +-----------+      |          +------+           |	   +-----------+
+   |  [veth1]  | ------ to_veth1-|  r1  |-to_veth2  ------ |  [veth2]  | 
+   +-----------+      |	         +------+           |	   +-----------+
+       ns1 	      +-----------------------------+		ns2
+  
+
+Topology construction
+**********************
+
+First we create the namespaces, we add the veths that are connected in an appropriate way and on each of them a network is configured and finally the default gateway is also added.
+
+
+::
+
+  for i in `seq 1 2`;
+	do
+		sudo ip netns add ns${i}
+		sudo ip link add veth${i}_ type veth peer name veth${i}
+		sudo ip link set veth${i}_ netns ns${i}
+		sudo ip netns exec ns${i} ip link set dev veth${i}_ up
+		sudo ip link set dev veth${i} up
+		sudo ip netns exec ns${i} ifconfig veth${i}_ 10.0.${i}.1/24
+		sudo ip netns exec ns${i} route add default gw 10.0.${i}.254 veth${i}_
+	done
+
+
+In another terminal just run polycube using ``sudo polycubed``.
+Regarding the concept of shadow: a shadow cube (only a standard cube can be shadow) is associated with a Linux network namespace. 
```

Comment:
I would reduce this text and make a reference to the concept of the shadow cube.

Reply:
Done!